packages/js/product-editor/src/blocks/generic/collapsible/edit.tsx

```diff
@@ -30,3 +35,63 @@ export function Edit( {
 		</div>
 	);
 }
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+const areAllBlocksInvisible = ( blocks: BlockInstance[], context: any ) => {
+	return blocks.every( ( block ) => {
+		if (
+			! block.attributes?._templateBlockHideConditions ||
+			! Array.isArray( block.attributes?._templateBlockHideConditions )
+		) {
+			return false;
+		}
+		return block.attributes._templateBlockHideConditions.some(
+			( condition ) => evaluate( condition.expression, context )
+		);
+	} );
+};
```

Comment:
cc: @mattsherman I wonder if we could re-visit this and optimize this logic a bit, or also consider a slightly different approach. This feels like a lot of logic for blocks that render nested blocks to figure out if they should still be shown or not.
Maybe this should be handled as part of `registerWooBlockType` function? Or we move to actually removing the blocks on the fly so that we can just rely on the `innerBlocks`

Reply:
I didn't add this for all `woocommerce/product-collapsible` blocks because I'm having to make assumptions about the innerBlocks. For example, in this case, there is a [woocommerce/product-section](https://github.com/woocommerce/woocommerce/pull/43228/files#diff-984a5e05f019cbf7de89ac2a388bf9569852d7b5c8f05ce6684ba95f5d6029bcR78) as an innerBlock. That's the case for `product-inventory-advanced`, but not for `product-pricing-advanced`. 

Overall I'm not happy with the solution, but I can't find a better way of achieving this.